apps/settings/sub_menu/about_controller.cpp

```diff
@@ -113,10 +113,22 @@ bool AboutController::handleEvent(Ion::Events::Event event) {
         MessageTableCellWithBuffer * myCell = (MessageTableCellWithBuffer *)m_selectableTableView.selectedCell();
         char batteryLevel[5];
         if(strchr(myCell->accessoryText(), '%') == NULL){
-          int batteryLen = Poincare::Integer((int) ((Ion::Battery::voltage() - 3.6) * 166)).serialize(batteryLevel, 5);
-          batteryLevel[batteryLen] = '%';
-          batteryLevel[batteryLen+1] = '\0';
-        }else{
+          #ifdef DEVICE
+            if(!Ion::Battery::isCharging() && Poincare::Integer((int) ((Ion::Battery::voltage() - 3.6) * 166)) <= 100) {
+              int batteryLen = Poincare::Integer((int) ((Ion::Battery::voltage() - 3.6) * 166)).serialize(batteryLevel, 5);
+              batteryLevel[batteryLen] = '%';
+              batteryLevel[batteryLen+1] = '\0';
+            } else {
+                myCell->setAccessoryText("100%");
+              } 
+              return true;
+            }
```

Comment:
This bracket should be removed, because it is closed at line 131.
```suggestion
```

Reply:
yes thanks sorry I was eating lol I will modify the error and for the separation between device and simulator I agree with you